### PR TITLE
For #1745 - Allow trackers from Allow List

### PR DESCRIPTION
**Pull request by ekager**
New behavior: Toggling Tracking Protection in Quick Settings now adds/removes a domain (host of URL) to Exceptions List. If a site is on the Exceptions List, TP will be toggled off when URI is requested -> `AppRequestInterceptor` checks if the host is an exception and will turn on/off TP for the current engine session accordingly.

There are probably some operations that we could complete on a different thread or cache for performance enhancements, I am open to suggestions!! I wanted to get the bulk of the work done since I will be gone at I/O and mostly unavailable this week :)

### Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

**Comment by ekager** [comment]
I also sent @AmyYLee an APK for any UX feedback needed at this step.

File: app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsFragment.kt [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    private lateinit var exceptionsComponent: ExceptionsComponent
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        job = Job()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        (activity as AppCompatActivity).title = getString(R.string.preference_exceptions)
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/exceptions/ExceptionsFragment.kt]: FYI: This cast isn't strictly necessary and may get in the way of testing with Robolectric later.

File: app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt [hunk 2]
```diff
@@ -6,43 +6,66 @@ package org.mozilla.fenix.settings
...
-        (activity as AppCompatActivity).supportActionBar?.show()
-    }
-
     override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
         setPreferencesFromResource(R.xml.tracking_protection_preferences, rootKey)
     }
 
     override fun onResume() {
         super.onResume()
+        (activity as AppCompatActivity).title = getString(R.string.preferences_tracking_protection)
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/settings/TrackingProtectionFragment.kt]: Unneeded cast

File: app/src/main/res/layout/component_exceptions.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<LinearLayout
```
Comment (colintheshots) [inline app/src/main/res/layout/component_exceptions.xml]: Do we need the outer layout? Extra view hierarchy has a real cost.

File: app/src/main/res/layout/delete_exceptions_button.xml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (colintheshots) [inline app/src/main/res/layout/delete_exceptions_button.xml]: Do we need the outer layout?

File: app/src/main/res/layout/exceptions_description.xml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- This Source Code Form is subject to the terms of the Mozilla Public
+   - License, v. 2.0. If a copy of the MPL was not distributed with this
+   - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (colintheshots) [inline app/src/main/res/layout/exceptions_description.xml]: Do we need the outer layout?

Comment (ekager) [reply app/src/main/res/layout/exceptions_description.xml]: Nope :)

File: app/src/main/res/layout/delete_exceptions_button.xml (hunk 4 above)
Comment (ekager) [reply app/src/main/res/layout/delete_exceptions_button.xml]: No

File: app/src/main/res/layout/component_exceptions.xml (hunk 3 above)
Comment (ekager) [reply app/src/main/res/layout/component_exceptions.xml]: I think we do need this one? This is how the other components are created, passing a wrapper view that the component's UI views are added to

Comment (colintheshots) [reply app/src/main/res/layout/component_exceptions.xml]: Sounds fine to me. Just had to check.

**Pull request merged**